http-client/src/test/java/io/airlift/http/client/TestHttpUriBuilder.java

```diff
@@ -348,10 +349,28 @@ public void testEncodesPath()
         assertEquals(uri.toASCIIString(), "http://www.example.com/%60%23%25%5E%7B%7D%7C%5B%5D%3C%3E%3F%C3%A1%C3%A9%C3%AD%C3%B3%C3%BA");
     }
 
-    @Test(expectedExceptions = IllegalStateException.class, expectedExceptionsMessageRegExp = ".*scheme.*")
-    public void testVerifiesSchemeIsSet()
+    @Test
+    public void testVerifyOnBuild()
     {
-        uriBuilder().build();
+        assertThatThrownBy(() -> uriBuilder().build())
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessage("scheme has not been set");
+
+        assertThatThrownBy(() -> uriBuilder().scheme("http").build())
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessage("host has not been set");
+    }
+
+    @Test
+    public void testVerifyOnCreate()
+    {
+        assertThatThrownBy(() -> uriBuilderFrom(URI.create("./foo")))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("URI does not have a scheme: ./foo");
+
+        assertThatThrownBy(() -> uriBuilderFrom(URI.create("http://test_foo/abc")))
```

Comment:
Add a comment why `test_foo` is not considered a host

Reply:
Done